src/services/UserService.ts

```diff
@@ -59,7 +59,7 @@ export class UserService {
             const newUser = await this.userRepository.save(user);
             return newUser;
         } else {
-            return tempUser;
+            return undefined;
```

Comment:
This whole check with `tempUser` can be removed given that these columns are already Unique, in which case the DB will throw an error if a duplicate User is inserted (or just a certain field). Both the `username` and `publickey` are already set to unique. This `tempUser` check is the one causing errors for you... so just remove it and catch the DB error.

Reply:
I tried without it and if the username with the same public key is in the database it does overwrite it and rewrite it to the database

Reply:
After a discussion about this edge case the person could log in anyway if they know the username and the public key so the extra check was removed